docs/guides/self_hosting.md

```diff
@@ -0,0 +1,117 @@
+# [Experimental] Self Hosting
+**This guide is currently experimental. Instructions are subject to change as we improve self-hosting capabilities.**
```

Comment:
Let's make the warning more explicit:

**Self hosting support is currently highly experimental. Attempt at your own risk. Self-hosting instructions are subject to change as we improve self-hosting support.**